File: Automation_App_Frontend/lib/features/email_versand/domain/entities/signatur_bild.dart
import 'package:equatable/equatable.dart';

/// Ein Bild, das in der formatierten Signatur steckt: Kanzleilogo,
/// Zertifikatssiegel, das animierte Werbebild (§4.7).
///
/// Die Größe steht dabei, weil genau sie die Frage entscheidet, die der Anwalt
/// je Mail beantwortet: Ein paar Kilobyte Logo gehen immer mit, ein GIF von
/// mehreren Megabyte nicht unter jede Nachricht.
class SignaturBild extends Equatable {
  final String dateiname;
  final int bytes;

  /// Der Inhalt des Bildes in Kurzform, wie der Dienst ihn meldet — gehört an
  /// die Adresse, unter der die Vorschau es lädt.
  ///
  /// Der Name allein taugt dafür nicht: Outlook nennt das erste Bild **jeder**
  /// Signatur `image001.png`. Flutter hebt geladene Bilder je Adresse auf, und
  /// weil die sich beim Signaturwechsel nicht änderte, zeigte die Vorschau
  /// weiter das Logo der vorigen — bis zum Neustart der App (behoben am
  /// 04.09.2026). Leer heißt: Der Dienst konnte sie nicht bilden; dann bleibt
  /// es beim alten Verhalten.
  final String marke;

  const SignaturBild({
    required this.dateiname,
    required this.bytes,
    this.marke = '',
  });

  factory SignaturBild.fromJson(Map<String, dynamic> json) => SignaturBild(
    dateiname: json['dateiname'] as String? ?? '',
    bytes: (json['bytes'] as num?)?.toInt() ?? 0,
    marke: json['marke'] as String? ?? '',
  );

  @override
  List<Object?> get props => [dateiname, bytes, marke];
}

File: AutomationService/AutomationService/Features/EmailVersand/Domain/Services/SignaturBild.cs
namespace AutomationService.Features.EmailVersand.Domain.Services;

/// <summary>
/// Ein Bild, das in der formatierten Signatur steckt: Kanzleilogo,
/// Zertifikatssiegel, das animierte Werbebild (§4.7).
///
/// Die Größe steht bewusst dabei. Genau sie ist der Grund, warum solche Bilder
/// nicht unter jede Mail gehören: Ein animiertes GIF von mehreren Megabyte
/// wächst mit jeder Nachricht mit, und der Anwalt entscheidet je Mail, ob es
/// mitgeht.
/// </summary>
/// <param name="Dateiname">Der blanke Name, unter dem es in der Ablage liegt.</param>
/// <param name="Bytes">Wie schwer es wiegt.</param>
/// <param name="Marke">
/// Sein Inhalt in Kurzform (<see cref="SignaturMarke"/>). Der Name allein
/// unterscheidet zwei Logos nicht — Outlook nennt beide <c>image001.png</c>.
/// </param>
public sealed record SignaturBild(string Dateiname, long Bytes, string Marke);
